package/src/services/check-parser/parser.ts

```diff
@@ -106,11 +55,9 @@ export class Parser {
   * We can find all of the files we need to run the check by traversing this graph.
   * In this implementation, we use breadth first search.
   */
-    const missingFiles: string[] = []
-    const unsupportedNpmDependencies: UnsupportedNpmDependencies[] = []
-    const parseErrors: ParseError[] = []
-    const dependencies = new Set<string>()
-    dependencies.add(entrypoint)
+    const collector = new Collector(entrypoint)
+    // Stop adding entrypoint as dependency later on
```

Comment:
What do you mean by `Stop adding entrypoint as dependency later on`? Is this a TODO?

We'll need to do something to handle circular dependencies on the `entrypoint`, since you can have circular dependencies in JS. I think that just adding the `entrypoint` to the dependencies and removing it later is a nice solution.

Reply:
Ok, sounds good 👍